### PR TITLE
[wip] V8

### PR DESCRIPTION
**Pull request by tmcw**
Implements breaking changes of https://github.com/mapbox/mapbox-gl-style-spec/pull/264

* [x] Rename symbol-min-distance to symbol-spacing. 
* [x] Flip coordinate order of video coordinates
* [x] Support font stacks as arrays
* [x] background-image ⇢ background-pattern
* [x] line-image ⇢ line-pattern
* [x] fill-image ⇢ fill-pattern
* [x] Rename video url to urls.
* [x] ImageSource https://github.com/mapbox/mapbox-gl-js/issues/700
* [x] Color-operations: needs to be merged from https://github.com/mapbox/mapbox-gl-js/pull/1158 into this branch
* [x] Circle type: we've decided that we really want this in this revision. https://github.com/mapbox/mapbox-gl-js/issues/1240
  * [x] circle offset

### CODE REVIEW

File: debug/bright-v7-types.json [hunk 1]
```diff
@@ -0,0 +1,1324 @@
...
+    "@building_shadow": { "type": "color", "value": "#dfdbd7" },
+    "@building_color_transition": {
+        "type": "color",
+        "value": {
+          "base": 1,
+          "stops": [[15.5, "#f2eae2"], [16, "#dfdbd7"]]
+        }
+    },
+    "@building_opacity": {
+        "type": "color",
```
Comment (lbud) [inline debug/bright-v7-types.json]: Should this be "opacity?"

File: debug/bright-v7-types.json [hunk 2]
```diff
@@ -0,0 +1,1324 @@
...
+    "@path": { "type": "color", "value": "#cba" },
+    "@rail": { "type": "color", "value": "#bbb" },
+    "@text": { "type": "color", "value": "#334" },
+    "@text_halo": { "type": "color", "value": "rgba(255 },255 },255 },0.8)" },
+    "@marine_text": { "type": "color", "value": "#74aee9" },
+    "@marine_text_halo": { "type": "color", "value": "rgba(255 },255 },255 },0.7)" },
+    "@poi_text": { "type": "color", "value": "#666" },
+    "@poi_text_halo": { "type": "color", "value": "#ffffff" },
+    "@maki": { "type": "color", "value": "#666" },
+    "@point_translate": { "type": "dasharray", "value": [0, -30] },
```
Comment (lbud) [inline debug/bright-v7-types.json]: Hm, spec (https://github.com/mapbox/mapbox-gl-style-spec/pull/264/files#diff-8330e164d8515979491567746e4dc5c7R25) looks like `dash-array`, not `dasharray`, right?

File: debug/bright-v7-types.json [hunk 3]
```diff
@@ -0,0 +1,1324 @@
+{
+  "version": 7,
+  "name": "Bright",
+  "constants": {
+    "@name": { "type": "field", "value": "{name_en}" },
```
Comment (lbud) [inline debug/bright-v7-types.json]: `field` or `field-template`?

Comment (tmcw) [inline debug/bright-v7-types.json]: Yeah - this style was slowly slowly manually updated. If you aren't relying on it, I'd like to remove it from this PR and rely on gl-style-migrate for our testing material.

Comment (lbud) [inline debug/bright-v7-types.json]: Yep, not relying on it, just visually checking what all the types should look like — fire away.

**Comment by bsudekum** [comment]
:tada:

**Comment by tmcw** [comment]
:raised_hands: !!

**Pull request merged**